internal/cluster/transition.go

```diff
@@ -482,6 +483,21 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 				reason, params.eventHandler, extra...)
 		}
 
+		//report cluster install duration metrics in case of error. Cancel and Installed cases are
+		//treated separately in CancelInstallation and CompleteInstallation respectively
+		if sCluster.srcState != swag.StringValue(sCluster.cluster.Status) {
```

Comment:
```suggestion
		if funk.ContainsString(["error","pending-user-action], sCluster.cluster.Status)
				params.metricApi.ClusterInstallationFinished(params.ctx, sCluster.cluster.Status, sCluster.srcState,
					sCluster.cluster.OpenshiftVersion, *sCluster.cluster.ID, sCluster.cluster.EmailDomain,
					sCluster.cluster.InstallStartedAt)
```

Reply:
Here's the reason is important and not the state. RefreshCluster is called with other reasons on the same statuses

Reply:
After some thought I accept the proposal (state should be taken from the updated cluster though. See Fix